knowledge_storm/rm.py

```diff
@@ -1093,3 +1094,109 @@ def forward(
             collected_results.append(r)
 
         return collected_results
+
+class AzureAISearch(dspy.Retrieve):
+    """Retrieve information from custom queries using Azure AI Search. General Documentation can be found at: https://learn.microsoft.com/en-us/azure/search/search-create-service-portal. Python Documentation and examples can be found at https://learn.microsoft.com/en-us/python/api/overview/azure/search-documents-readme?view=azure-python. Requires pip install azure-search-documents"""
```

Comment:
Add a line break before `General Documentation`.

Also, I found this PR fails the Python format check. Check out https://github.com/stanford-oval/storm/blob/main/CONTRIBUTING.md to see how to abide by the format rule; otherwise, the merging will be blocked.

Reply:
added the pre-commit hook locally and should be passing with the latest push.